### PR TITLE
fix an edge case where overflow check in QuoRem is unreachable

### PR DESCRIPTION
**Pull request by frankli0324**
pretty straightforward, int32 overflows before the overflow check

### CODE REVIEW

**Comment by frankli0324** [comment]
it's also worth noting that https://github.com/shopspring/decimal/blob/master/decimal_test.go#L2164C1-L2171 didn't include tests for that

File: decimal.go [hunk 1]
```diff
@@ -565,7 +565,7 @@ func (d Decimal) QuoRem(d2 Decimal, precision int32) (Decimal, Decimal) {
 		panic("decimal division by 0")
 	}
 	scale := -precision
-	e := int64(d.exp - d2.exp - scale)
+	e := int64(d.exp) - int64(d2.exp) - int64(scale)
```
Comment (mwoss) [inline decimal.go:568]: Thanks for catching that! :3 LGTM!
Could you add a test case for this edge case? If you don't have time I can work on that.

**Review by mwoss — APPROVED** [review]

File: decimal.go (hunk 1 above)
Comment (frankli0324) [reply decimal.go:568]: Sorry I do have no time on this. Could you please handle it?

Comment (mwoss) [reply decimal.go:568]: Sure will do, no problem :3
